exp/tools/tool.go

```
@@ -0,0 +1,17 @@
+// Package tools defines the types for tools to be used by the llms agents.
+package tools
+
+// Tool is a tool for the llm agent to interact with diferent application.
+type Tool struct {
+	Name        string
+	Description string
+	Run         func(string) string
```

This should return an optional error

---

The Run function right ? Like ```go
func(string)(string,error)
```

---

Done x2 :)